components/webdriver_server/lib.rs

```
@@ -840,6 +857,49 @@ impl Handler {
+        let browsing_context_id = self.session()?.browsing_context_id;
+
+        match self.load_status_receiver.try_recv() {
+            Ok(WebDriverLoadStatus::NavigationStarted) => {
+                // Check if the navigation is approved or not with constellation
+                match self.load_status_receiver.recv() {
+                    Ok(WebDriverLoadStatus::Loading) => {
+                        self.wait_for_document_ready_state()?;
+                        Ok(WebDriverResponse::Void)
+                    },
+                    Ok(WebDriverLoadStatus::Canceled) => Ok(WebDriverResponse::Void),
+                    Ok(_) => unreachable!("Unexpected load status received"),
+                    Err(_) => Ok(WebDriverResponse::Void),
+                }
+            },
+            Ok(WebDriverLoadStatus::NavigationHashChanged) => {
+                let (sender, receiver) = ipc::channel().unwrap();
+                self.send_message_to_embedder(WebDriverCommandMsg::ScriptCommand(
+                    browsing_context_id,
+                    WebDriverScriptCommand::WaitCurrentDomEvents(sender),
+                ))?;
+                let _ = wait_for_script_response(receiver)?;
+                Ok(WebDriverResponse::Void)
+            },
+            Ok(WebDriverLoadStatus::Loading) |
+            Ok(WebDriverLoadStatus::Canceled) |
+            Ok(WebDriverLoadStatus::Complete) => {
+                // TODO: We may ignore Complete?
+                unreachable!("Unexpected load status received")
+            },
+            Ok(WebDriverLoadStatus::Timeout) => {
+                // If the load status is timeout, we should return an error
+                Err(WebDriverError::new(
+                    ErrorStatus::Timeout,
+                    "Navigation timed out",
+                ))
+            },
+            // Empty channel means no navigation started
+            Err(_) | Ok(WebDriverLoadStatus::Blocked) => Ok(WebDriverResponse::Void),
```

Same here. It's weird that we are ignoring all errors.